ros/src/tl_detector/tl_detector.py

```diff
@@ -170,41 +174,28 @@ def process_traffic_lights(self):
             int: index of waypoint closest to the upcoming stop line for a traffic light (-1 if none exists)
             int: ID of traffic light color (specified in styx_msgs/TrafficLight)
 
-        """
-        light = None
-
-        # List of positions that correspond to the line to stop in front of for a given intersection
-        # stop_line_positions = self.config['stop_line_positions']
-        # if(self.pose):
-        #     car_position = self.get_closest_waypoint(self.pose.pose)
-
-        # Find the closest visible traffic light (if one exists)
-        #TODO maybe only execute this if a light was actually seen in a camera image
-        return self.get_closest_light(self.pose.pose)
-
-    def get_closest_light(self, pose):
-        """Identifies the closest traffic light to the given position.
-
-        Args:
-            post (Pose): position to match a light to
-
-        int: index of the closest light in self.lights.
         """
         if not self.lights :
             return -1, TrafficLight.UNKNOWN
 
-        x = pose.position.x
-        y = pose.position.y
-        wp_idx = self.get_closest_waypoint_xy(x, y)
+        if(self.pose):
+            car_wp_idx = self.get_closest_waypoint(self.pose.pose)
 
-        for l_idx in self.lights_red_2d:
-            if l_idx > wp_idx:
-                return l_idx, TrafficLight.RED
+            for l_idx in self.lights_red_idxs:
+                stop_x, stop_y = self.stop_line_positions[l_idx]
+                # Car should stop *before* the stop line.
```

Comment:
@goforthecheez tl_detector should not define where car stops - it should just tell where the line is
Waypoint updater is responsible for doing waypoint planning

Reply:
What I mean here is that the "- 1" is needed. get_closest_waypoint_xy returns the closest waypoint *after* the (x, y) stp line position passed in. But if the car is to stop before the line, it function needs to report the closest waypoint just before this position. All fine-tuning has always been for the WPU to decide.